src/main/java/com/cqse/teamscaleupload/ReportPatternManager.java

```diff
@@ -13,23 +14,49 @@
 import java.util.stream.Collectors;
 
 /**
- * This class manages the given file formats and
- * corresponding report file patterns.
+ * This class provides the functionality to resolve the report file patterns and formats
+ * from the command line to a map from report formats to actual files which can then be
+ * used for the upload to Teamscale.
  */
 public class ReportPatternManager {
 
-
-    /** A map from file formats to corresponding report file patterns. */
-    private final Map<String, Set<String>> formatToFileNames = new HashMap<>();
-
     /** Expected pattern for formats in the input file, matches e.g. "[vs_coverage]" */
     private static final Pattern FORMAT_PATTERN = Pattern.compile("\\[(\\w+)\\]");
 
     /**
-     * Reads all file patterns which are specified in the given input file
-     * and adds them to {@link #formatToFileNames} for the respective format.
+     * Returns a map from file formats to corresponding report files.
+     *
+     * <ol>
+     *     <li>Reads file patterns and formats from the input file</li>
+     *     <li>Reads file patterns for the format specified on the command line</li>
+     *     <li>Resolves all file patterns from the previous steps to actual files.</li>
+     * </ol>
      */
-    public void addReportFilePatternsFromInputFile(Path input) throws IOException {
+    public static Map<String, Set<File>> resolveInputFilePatters(Path inputFile, List<String> filePatterns, String format)
+            throws IOException, AgentOptionParseException {
+        Map<String, Set<String>> formatToFilePatterns = new HashMap<>();
+
+        if (inputFile != null) {
+            formatToFilePatterns = addReportFilePatternsFromInputFile(inputFile);
+        }
+        addFilePatternsForFormatOnCommandLine(formatToFilePatterns, filePatterns, format);
+
+        // Resolve all file patterns to the actual files for all given formats.
```

Comment:
extract this part to its own method? inline comments for me are hints that this is something that should be encapsulated in a funciton with a speaking name instead